dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -0,0 +1,115 @@
+#include <winrt/Windows.ApplicationModel.Core.h>
+#include <winrt/Windows.Storage.h>
+#include <winrt/base.h>
+#include "ToastActivationCallback.h"
+#include "externs.h"
+#include "ToastNotificationUtility.h"
+#include <frameworkudk/pushnotifications.h>
+
+static winrt::event<winrt::Windows::Foundation::EventHandler<winrt::Microsoft::Windows::ToastNotifications::ToastActivatedEventArgs>> g_toastHandlers;
+
+static int g_toastHandlerCount = 0;
+
+winrt::event<winrt::Windows::Foundation::EventHandler<winrt::Microsoft::Windows::ToastNotifications::ToastActivatedEventArgs>>& GetToastHandlers()
+{
+    return g_toastHandlers;
+}
+
+int& GetToastHandleCount()
+{
+    return g_toastHandlerCount;
+}
+
+namespace winrt
+{
+    using namespace winrt::Windows::UI;
+    using namespace winrt::Windows::Foundation;
+    using namespace Windows::ApplicationModel::Core;
+}
+
+namespace winrt::Microsoft::Windows::ToastNotifications::implementation
+{
+    static wil::unique_com_class_object_cookie s_toastcomActivatorRegistration;
+
+    winrt::Microsoft::Windows::ToastNotifications::ToastNotificationManager ToastNotificationManager::Default()
+    {
+        static auto toastNotificationManager{ winrt::make<ToastNotificationManager>() };
+        return toastNotificationManager;
+    }
+
+    void ToastNotificationManager::RegisterActivator(winrt::Microsoft::Windows::ToastNotifications::ToastActivationInfo const&  details)
+    {
+        THROW_HR_IF_MSG(E_INVALIDARG, s_toastcomActivatorRegistration, "ComActivator already registered.");
+
+        THROW_HR_IF(E_INVALIDARG, (details == nullptr));
+
+        std::wstring storedComActivatorString;
+        if (!AppModel::Identity::IsPackagedProcess())
+        {
+            storedComActivatorString = RegisterComActivatorGuidAndAssets(details);
+            // Remove braces around the guid string
+            storedComActivatorString = storedComActivatorString.substr(1, storedComActivatorString.size() - 2);
+        }
+
+        THROW_IF_FAILED(::CoRegisterClassObject(
+            AppModel::Identity::IsPackagedProcess() ? details.TaskClsid() : winrt::guid(storedComActivatorString),
+            winrt::make<ToastActivationCallbackFactory>().get(),
+            CLSCTX_LOCAL_SERVER,
+            REGCLS_MULTIPLEUSE,
+            &s_toastcomActivatorRegistration));
+
+        GetWaitHandleForArgs().create();
+    }
+
+    void ToastNotificationManager::UnregisterActivator()
+    {
+        THROW_HR_IF_MSG(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), !s_toastcomActivatorRegistration, "ComActivator not registered.");
+
+        s_toastcomActivatorRegistration.reset();
+
+        if (!AppModel::Identity::IsPackagedProcess())
+        {
+            std::wstring storedComActivatorString{ RegisterComActivatorGuidAndAssets(nullptr /* ToastActivationInfo */)};
+
+            UnRegisterComServer(storedComActivatorString);
+            UnRegisterAppIdentifierFromRegistry();
+
+            // TODO: Remove ToastGuid reference from LRP
```

Comment:
Seperate Task for this?

Reply:
@pmpurifoy  can you please let me know which task of yours that you are going to handle this?

Reply:
https://github.com/microsoft/WindowsAppSDK/pull/1923

This PR has the LRP remove toast function.